### PR TITLE
perf(redis): Stop using KEYS command in BakeStore

### PR DESCRIPTION
**Pull request by danopia**
Replace `jedis.keys` invocations with a kork `withKeyScan` invocation, like https://github.com/spinnaker/igor/pull/270 (instead of implementing a scan loop directly). Came up from Redis slowlog entries reaching dozens of ms. 

Note that I did not change out the other `jedisPool` usages in `RedisBackedBakeStore.groovy`. Replacing them introduced a bunch of these 'ambiguous call' warns:
Screen Shot 2019-04-22 at 9 56 18 AM

I was also unable to locate a `jedis.time()` signature for `getTimeInMilliseconds()`. The 'TIME' command is seemingly only exposed directly on BinaryClient.

For these reasons I only touched the 'keys' invocations in this initial commit.

### CODE REVIEW

File: rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/persistence/RedisBackedBakeStore.groovy [hunk 1]
```diff
@@ -453,29 +456,17 @@ class RedisBackedBakeStore implements BakeStore {
...
   public String deleteBakeByKey(String bakeKey) {
-    def jedis = jedisPool.getResource()
     def keyList = [bakeKey, "allBakes"]
-
-    jedis.withCloseable {
-      Set<String> incompleteBakesKeys = jedis.keys(allIncompleteBakesKeyPattern)
-
-      keyList += incompleteBakesKeys
-    }
+    keyList += this.scanIncompleteBakesKeys()
```
Comment (cfieber) [inline rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/persistence/RedisBackedBakeStore.groovy]: minor stylistic nit - you don't need the explicit `this` qualifier to call the scanIncompleteBakesKeys method, can just write this as
```groovy
keyList += scanIncompleteBakesKeys()
```

File: rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/persistence/RedisBackedBakeStore.groovy [hunk 2]
```diff
@@ -453,29 +456,17 @@ class RedisBackedBakeStore implements BakeStore {
...
-    def jedis = jedisPool.getResource()
     def updatedTimestampMilliseconds = timeInMilliseconds
     def keyList = [bakeKey, "allBakes"]
-
-    jedis.withCloseable {
-      Set<String> incompleteBakesKeys = jedis.keys(allIncompleteBakesKeyPattern)
-
-      keyList += incompleteBakesKeys
-    }
+    keyList += this.scanIncompleteBakesKeys()
```
Comment (cfieber) [inline rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/persistence/RedisBackedBakeStore.groovy]: same comment as above

File: rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/persistence/RedisBackedBakeStore.groovy [hunk 3]
```diff
@@ -489,15 +480,9 @@ class RedisBackedBakeStore implements BakeStore {
...
-    def jedis = jedisPool.getResource()
     def updatedTimestampMilliseconds = timeInMilliseconds
     def keyList = [bakeId, "allBakes"]
-
-    jedis.withCloseable {
-      Set<String> incompleteBakesKeys = jedis.keys(allIncompleteBakesKeyPattern)
-
-      keyList += incompleteBakesKeys
-    }
+    keyList += this.scanIncompleteBakesKeys()
```
Comment (cfieber) [inline rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/persistence/RedisBackedBakeStore.groovy]: same comment as above

File: rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/persistence/RedisBackedBakeStore.groovy (hunk 1 above)
Comment (danopia) [reply rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/persistence/RedisBackedBakeStore.groovy]: Thanks! I'm not sure I fully registered that this is a Groovy file 🙈

**Pull request merged**